apple/MarkdownTextInputDecoratorView.mm

```diff
@@ -51,26 +48,55 @@ - (void)didMoveToWindow {
 
 #ifdef RCT_NEW_ARCH_ENABLED
   react_native_assert([view isKindOfClass:[RCTTextInputComponentView class]] && "Previous sibling component is not an instance of RCTTextInputComponentView.");
-  _textInput = (RCTTextInputComponentView *)view;
-  _backedTextInputView = [_textInput valueForKey:@"_backedTextInputView"];
+  RCTTextInputComponentView *textInputComponentView = (RCTTextInputComponentView *)view;
+  UIView<RCTBackedTextInputViewProtocol> *backedTextInputView = [textInputComponentView valueForKey:@"_backedTextInputView"];
 #else
   react_native_assert([view isKindOfClass:[RCTBaseTextInputView class]] && "Previous sibling component is not an instance of RCTBaseTextInputView.");
-  _textInput = (RCTBaseTextInputView *)view;
-  _backedTextInputView = _textInput.backedTextInputView;
+  RCTBaseTextInputView *baseTextInputView = (RCTBaseTextInputView *)view;
+  UIView<RCTBackedTextInputViewProtocol> *backedTextInputView = baseTextInputView.backedTextInputView;
 #endif /* RCT_NEW_ARCH_ENABLED */
 
   _markdownUtils = [[RCTMarkdownUtils alloc] init];
   react_native_assert(_markdownStyle != nil);
   [_markdownUtils setMarkdownStyle:_markdownStyle];
 
-  [_textInput setMarkdownUtils:_markdownUtils];
-  if ([_backedTextInputView isKindOfClass:[RCTUITextField class]]) {
-    RCTUITextField *textField = (RCTUITextField *)_backedTextInputView;
-    _adapter = [textField valueForKey:@"textInputDelegateAdapter"];
-    [_adapter setMarkdownUtils:_markdownUtils];
-  } else if ([_backedTextInputView isKindOfClass:[RCTUITextView class]]) {
-    _textView = (RCTUITextView *)_backedTextInputView;
-    [_textView setMarkdownUtils:_markdownUtils];
+  if ([backedTextInputView isKindOfClass:[RCTUITextField class]]) {
+    _textField = (RCTUITextField *)backedTextInputView;
+    
+    // make sure `adjustsFontSizeToFitWidth` is disabled, otherwise formatting will be overwritten
+    react_native_assert(_textField.adjustsFontSizeToFitWidth == NO);
+    
+    _markdownTextFieldObserver = [[MarkdownTextFieldObserver alloc] init];
+    _markdownTextFieldObserver.markdownUtils = _markdownUtils;
+    _markdownTextFieldObserver.textField = _textField;
+    _markdownTextFieldObserver.active = YES;
```

Comment:
Is there a scenario where it would be inactive (ignoring the avoiding recursion part)? If not, shouldn't it be active by default?

Reply:
Moved to nice constructors added in 40abeba